mtz-update-password.html

```diff
@@ -394,14 +440,26 @@
       __updateStrengthMeter(value) {
 
         //do not run strength meter when its hidden
-        if (this.hideStrengthMeter) return;
+        if (!this.showStrengthMeter) return;
 
-        const result = zxcvbn(value);
-        // Update the password strength meter
-        this.__passwordStrengthMeterValue = result.score;
+        this.importLazyGroup('zxcvbn').then(() => {
+          const result = zxcvbn(value);
+          // Update the password strength meter
+          this.__passwordStrengthMeterValue = result.score;
 
-        // Update the text indicator
-        this.__passwordStrengthTextValue = value !== '' ? `Strength: ${this.strength[result.score]}` : '';
+          // Update the text indicator
+          this.__passwordStrengthTextValue = value !== '' ?
+            `Strength:&nbsp;<span class="result-color${result.score}">${this.strength[result.score]}</span>` : '';
+        });
+      },
+      /**
+       * set password type when the reveal icon is clicked
+       * @private
+       *
+       * @param {Boolean} checked
+       */
+      __setPasswordType(checked) {
+        this.$.password.type = checked ? 'text' : 'password';
```

Comment:
Instead of using DOM manipulation directly, change this to `return checked ? 'text' : 'password';` and update the template above to use `type$="[[__setPasswordType(__checked)]]"`, you can then drop the property observer.